src/github3/checks.py

```diff
@@ -0,0 +1,416 @@
+        self.id = run['id']
+        self.external_id = run['external_id']
+        self.app = CheckApp(run['app'], self)
+        self.check_suite = run['check_suite']['id']
+        # self.output = CheckRunOutput(run['output'], self)
+        self.output = run['output']  # TODO: turn into an object
+
+    def _repr(self):
+        return '<{s.class_name} [{s.name}:{s.status}]>'.format(s=self)
+
+    @decorators.requires_app_installation_auth
+    def update(self, name=None, details_url=None, external_id=None,
+               started_at=None, status=None, conclusion=None,
+               completed_at=None, output=None, actions=None):
+        """Update this check run.
+
+        All parameters are optional.
+
+        :param str name:
+            (optional), new name of the check
+        :param str details_url:
+            (optional), The URL of the integrator's site that has the full
+            details of the check
+        :param str external_id:
+            (optional), A reference for the run on the integrator's system
+        :param str started_at:
+            (optional), ISO 8601 time format: YYYY-MM-DDTHH:MM:SSZ
+        :param str status:
+            (optional), ('queued', 'in_progress', 'completed')
+        :param str conclusion:
+            (optional), Required if you provide 'completed_at', or a
+            'status' of 'completed'. The final conclusion of the check.
+            ('success', 'failure', 'neutral', 'cancelled', 'timed_out',
+            'action_required')
+        :param str completed_at:
+            (optional), Required if you provide 'conclusion'. ISO 8601 time
+            format: YYYY-MM-DDTHH:MM:SSZ
+        :param dict output:
+            (optional), key-value pairs representing the output. Format:
+            {'title': 'string', 'summary', 'text, can be markdown', 'text':
+            'text, can be markdown', 'annotations': [{}], 'images': [{}]}
+        :param array actions:
+            (optiona), array of action objects. Object format is:
+            {'label': 'text', 'description', 'text', 'identifier', 'text'}
+        :returns:
+            True if successful, False otherwise
+        :rtype:
+            bool
+        """
+        # TODO: Clean output dict, actions array. Need a deep recursive clean
+        data = {'name': name, 'details_url': details_url, 'external_id':
+                external_id, 'started_at': started_at, 'status': status,
+                'conclusion': conclusion, 'completed_at': completed_at,
+                'output': output, 'actions': actions}
+        self._remove_none(data)
+        json = None
+
+        if data:
+            json = self._json(self._patch(
+                self._api, data=dumps(data),
+                headers=CheckSuite.CUSTOM_HEADERS), 200
+            )
+        if json:
+            self._update_attributes(json)
+            return True
+        return False
+
+    @decorators.requires_app_installation_auth
+    def rerequest(self):
```

Comment:
@omgjlk is this a leftover from a copy&paste? From the docs I can't see anything about re-requesting a CheckRun.

Reply:
Yes that's probably the case. I think via the `actions` object you may be able to expose a Rerequest action for a specific run, but that may go to the provider of the run rather than something the GitHub API does. This function here can be dropped.